libfeed/feed-store-provider.c

```diff
@@ -0,0 +1,1239 @@
+                                       individual_task_result_closure_new (all_tasks_results_closure_allocate_new_result_slot (all_tasks_closure),
+                                                                           all_tasks_closure));
+      else if (g_strcmp0 (interface_name, "com.endlessm.DiscoveryFeedArtwork") == 0)
+        append_stores_task_from_proxy (ka_proxy,
+                                       append_discovery_feed_artwork_from_proxy,
+                                       NULL,
+                                       cancellable,
+                                       individual_task_result_completed,
+                                       individual_task_result_closure_new (all_tasks_results_closure_allocate_new_result_slot (all_tasks_closure),
+                                                                           all_tasks_closure));
+      else if (g_strcmp0 (interface_name, "com.endlessm.DiscoveryFeedWord") == 0)
+        g_ptr_array_add (word_proxies, ka_proxy);
+      else if (g_strcmp0 (interface_name, "com.endlessm.DiscoveryFeedQuote") == 0)
+        g_ptr_array_add (quote_proxies, ka_proxy);
+    }
+
+  zipped_word_quote_proxies = zip_two_ptr_arrays (word_proxies,
+                                                  quote_proxies,
+                                                  (PtrArrayZipElementFactory) word_quote_proxy_bundle_new,
+                                                  (GDestroyNotify) word_quote_proxy_bundle_free);
+
+  for (i = 0; i < zipped_word_quote_proxies->len; ++i)
+    {
+      WordQuoteProxyBundle *bundle = g_ptr_array_index (zipped_word_quote_proxies, i);
+      append_discovery_feed_word_quote_from_proxies (bundle->word_proxy,
+                                                     bundle->quote_proxy,
+                                                     cancellable,
+                                                     individual_task_result_completed,
+                                                     individual_task_result_closure_new (all_tasks_results_closure_allocate_new_result_slot (all_tasks_closure),
+                                                                                         all_tasks_closure));
+    }
+}
+
+typedef struct _UnorderedResultsFromQueriesData
+{
+  EosDiscoveryFeedGjsStyleGDBusCallback callback;
+  gpointer                              user_data;
+} UnorderedResultsFromQueriesData;
+
+static UnorderedResultsFromQueriesData *
+unordered_results_from_queries_data_new (EosDiscoveryFeedGjsStyleGDBusCallback callback,
+                                         gpointer                              user_data)
+{
+  UnorderedResultsFromQueriesData *data = g_new0 (UnorderedResultsFromQueriesData, 1);
+
+  data->callback = callback;
+  data->user_data = user_data;
+
+  return data;
+}
+
+static void
+unordered_results_from_queries_data_free (UnorderedResultsFromQueriesData *data)
+{
+  g_free (data);
```

Comment:
We could pass `g_free` directly to any GDestroyNotify param, just had this wrapper here to make it explicit.